spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DontAssertInstanceofInTests.java

```diff
@@ -0,0 +1,76 @@
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.BytecodeScanningDetector;
+
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.AnnotationEntry;
+import org.apache.bcel.classfile.Code;
+import org.apache.bcel.classfile.Method;
+
+/**
+ * A detector that checks for lines in JUnit tests that look like `assertTrue(object instanceof Class)` and discourages them.
+ *
+ * It may be more useful to observe error messages from bad casts, as this may reveal more information regarding the
+ * circumstances of the exception than a "false is not true" assert message.
+ *
+ * This detector reports bugs of the type `JUA_DONT_ASSERT_INSTANCEOF_IN_TESTS`.
+ */
+public class DontAssertInstanceofInTests extends BytecodeScanningDetector {
+    private boolean isTest;
+
+    private BugInstance currBug = null;
+
+    BugReporter bugReporter;
+
+    public DontAssertInstanceofInTests(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    @Override
+    public void visitMethod(Method obj) {
+        isTest = false;
+        for (AnnotationEntry entry : obj.getAnnotationEntries()) {
+            String entryType = entry.getAnnotationType();
+            isTest |= "Lorg/junit/Test;".equals(entryType);
+        }
+    }
+
+    @Override
+    public void visitCode(Code obj) {
+        if (isTest)
+            super.visitCode(obj);
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
```

Comment:
I'll confirm that we need to call `super.sawOpcode(seen)` or not.

Reply:
I noticed that the super method was empty, so I didn't bother.

Reply:
confirmed. thanks